uber/templates/regform.html

```diff
@@ -52,9 +51,22 @@
     });
 
     {% if not attendee.is_new %}
-        while (window.BADGE_TYPES && BADGE_TYPES.options[0].extra < {{ attendee.amount_extra }}) {
+        // This removes any badge levels lower than the attendee has purchased already
+        while (window.BADGE_TYPES && BADGE_TYPES.options[0] && BADGE_TYPES.options[0].extra < {{ attendee.amount_extra }}) {
             BADGE_TYPES.options.splice(0, 1);
         }
+
+        // This is the same idea, except it disables the kick-in buttons and then hides them
```

Comment:
(Getting caught up on my code reviews.)

$(this).val() is going to return a string, but $.val('amount_extra') is going to return an integer. This will happen to work in most situations but will fail in others. Javascript will automatically do type coercion here but that can be buggy, so it would be better to make sure that you're comparing integers to other integers.

Also, there's no reason to do the ``if`` check here, since it's always safe to call ``.each()`` on an empty jQuery set.

Finally, I'll note that the indentation of this function is off starting with the inner if statement.